### DISCUSSION TITLE
Web Worker - How to restart a failed worker

### DISCUSSION THREAD

**Discussion by rhurlbatt** [anvilistas/anvil-labs#111] [Q&A]
I have a timer that checks on the status of a Web Worker while the Web Worker is running. The timer queues up the next job for the Worker. If the Web Worker fails (due to no internet connection) it raises a 
`ExternalError: Error: TypeError: Failed to fetch`

What is the best way of killing the current web worker and restarting it to clear this error? If I don't I have to restart the app to achieve this and then the Worker continues on its merry course.
Does it require that a new Worker instance be created in its place?

**Comment by s-cork** [comment]
What happens when you just run the next queued task?
(Waiting until your back online)

**Comment by rhurlbatt** [comment]
It just keeps returning the same error... I will try build a minimal
example today if I don't have any joy but as I understand it if I set the
task to None or use the kill method and then give it a new task everything
should be ok (when back online)?

**Reply by s-cork** [reply]
After a task has retuned or raised then you shouldn't need to kill it since it's complete. 

You should just be able to launch a new task of the same function.

**Reply by rhurlbatt** [reply]
So I am fairly sure it is probably more my implementation than anything else...
To get this to clear the error and continue on, I had to set the `task = None` and then relaunch with a new task. With a fresh brain, I think this is probably the intended behaviour since we weren't actually clearing the failed task.

**Reply by s-cork** [reply]
Sounds about right. I'm guessing the logic is something along the lines of 

```python
if self.task is None:
  self.task = worker.launch_task('foo')
```

But if you don't clear your local task it'll never launch the new one. That's just my guess.

**Reply by rhurlbatt** [reply]
Yes but not in such an eloquent manner. Cool thanks for the help, I think this has got it working ok at the moment... I will see how it fairs over the next few days.
